### PR TITLE
Add powermax GO Client to package

### PR DESCRIPTION
**Pull request by shekhar-j**
# Description
Package GoClient Library with provider.

# GitHub Issues
# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
All 

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below
powermax_portgroup.portgroup_1: Creating...
powermax_portgroup.portgroup_1: Creation complete after 0s [id=tfacc_pg_test_1]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.


powermax_portgroup.portgroup_1: Destroying... [id=tfacc_pg_test_1]
powermax_portgroup.portgroup_1: Destruction complete after 1s

Destroy complete! Resources: 1 destroyed.



```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [ ] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by nikitajoshi1 — APPROVED** [review]

**Comment by forrestxia** [comment]
@shekhar-j One of the 6 checks failed for this PR - Check Go formatting, linting, vetting, can it be fixed?

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,8 @@ website/node_modules
 *.iml
 *.test
 *.iml
+powermax-go-client
+terraform-provider-powermax
```
Comment (forrestxia) [inline .gitignore:29]: Why ignoring terraform-provider-powermax?

Comment (shekhar-j) [reply .gitignore:29]: @forrestxia why do we need this to be pushed? we will be building it with the code.

**Pull request merged**